app/models/miq_task.rb

```diff
@@ -24,13 +24,30 @@ class MiqTask < ApplicationRecord
 
   before_validation :initialize_attributes, :on => :create
 
-  before_destroy :check_associations
+  before_destroy :check_active, :check_associations
 
   virtual_has_one :task_results
   virtual_attribute :state_or_status, :string, :arel => (lambda do |t|
     t.grouping(Arel::Nodes::Case.new(t[:state]).when(STATE_FINISHED).then(t[:status]).else(t[:state]))
   end)
 
+  def is_active?
+    if state == "Active" || !%w(Finished Waiting_to_start Queued).include?(state)
+      true
+    else
+      false
+    end
+  end
+
+  def check_active
+    if is_active?
```

Comment:
common Ruby practice advises predicate methods (ending in ?) NOT to be prefixed by is_ or has_  So, in this case, the following would read better:
```ruby
if active?
...
```